api/leetcode-data.js

```diff
@@ -0,0 +1,104 @@
+import { dailyGetLeetcodeData, weeklyGetLeetcodeData, getProblemSet } from "./leetcode/index.js";
+import logger from "../logging.js";
+import staticProblemSets from './leetcode/static/problemSets.json' assert { type: "json" };
+
+export class LeetcodeData {
+
+    constructor() {}
+
+    getProblemSetData = async (limit, skip) => {
+        const { data: { problemsetQuestionList: { questions } } } = await getProblemSet(limit, skip);
```

Comment:
this setup is a little dangerous, especially because we don't use typescript; someone in the future could change the return type in `getProblemSet` without noticing that they're breaking some code that it is assuming this data structure.

A slightly better approach is to get the `data` in an object as a whole, and then have the null check for each one of the nested fields, it would be something like this:

```js
const resposne = await getProblemSet(limit, skip);

if (!response?.data?.problemsetQuestionList?.questions) {
    logger.error("Unable to fetch problemsetQuestionList");
    return;
}

```

This is technically the same thing, but exception free

Reply:
Done, thanks!